# Superstore Sales Dataset Analysis

This notebook explores the Superstore dataset and answers the given questions using Python and Pandas.

### Dataset Loading and Preview

In [1]:
import pandas as pd 

df = pd.read_csv("Sample - Superstore.csv")

df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11-08-2016,11-11-2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11-08-2016,11-11-2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,06-12-2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10-11-2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10-11-2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


##### Dataset Preview

The first five rows of the dataset are displayed above. This helps in understanding the structure of the data and the available columns.The preview above displays the first five records of the dataset and helps in understanding its structure.

#### Q1. How many rows and columns does the dataset have?

In [2]:
df.shape

(9994, 21)

##### Answer:  

The dataset contains 9994 rows and 21 columns.

#### Q2. List all the column names in the dataset.

In [3]:
list(df.columns)

['Row ID',
 'Order ID',
 'Order Date',
 'Ship Date',
 'Ship Mode',
 'Customer ID',
 'Customer Name',
 'Segment',
 'Country',
 'City',
 'State',
 'Postal Code',
 'Region',
 'Product ID',
 'Category',
 'Sub-Category',
 'Product Name',
 'Sales',
 'Quantity',
 'Discount',
 'Profit']

##### Answer

The dataset contains 21 columns. The complete list of column names is displayed in the output above.

#### Q3. Check for missing values.

In [4]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

##### Answer

No missing values were found in the dataset. All 21 columns contain complete data.

#### Q4. What is the highest Sales value in the dataset?

In [5]:
df.loc[df['Sales'].idxmax()]

Row ID                                                        2698
Order ID                                            CA-2014-145317
Order Date                                               3/18/2014
Ship Date                                                3/23/2014
Ship Mode                                           Standard Class
Customer ID                                               SM-20320
Customer Name                                          Sean Miller
Segment                                                Home Office
Country                                              United States
City                                                  Jacksonville
State                                                      Florida
Postal Code                                                  32216
Region                                                       South
Product ID                                         TEC-MA-10002412
Category                                                Techno

The above output shows the complete record with the highest sales value.

In [6]:
df['Sales'].max()

np.float64(22638.48)

##### Answer

The highest sales value in the dataset is **22638.48**.

#### Q5. Which product Category appears the most number of times?

In [7]:
category_counts = df['Category'].value_counts()

most_category = category_counts.idxmax()
count = category_counts.max()

print("Most Frequent Category:", most_category)
print("Number of Occurrences:", count)

Most Frequent Category: Office Supplies
Number of Occurrences: 6026


##### Answer

The product category that appears the most number of times is **Office Supplies**.

It appears **6026 times** in the dataset.

#### Q6. What does this dataset seem to be about? Write 2-3 lines in your own words.

##### Answer

This dataset appears to contain sales records of a superstore. It provides details about customers, products, orders, and the revenue generated from sales.

The data can be used to study sales trends, compare product categories, and understand how the business is performing in different locations.


### Conclusion

The Superstore dataset contains 9994 records and 21 columns with no missing values. The highest sales value recorded is 22638.48, and Office Supplies is the most frequent product category. The dataset provides useful information for analyzing sales performance, customer behavior, and product trends.